src/extension.ts

```diff
@@ -1,398 +1,194 @@
+
+  const kicsDiagnosticCollection =
+    vscode.languages.createDiagnosticCollection(constants.extensionName);
+  const kicsStatusBarItem = vscode.window.createStatusBarItem(
+    vscode.StatusBarAlignment.Left
+  );
+  const kicsProvider = new KicsProvider(
+    context,
+    logs,
+    kicsStatusBarItem,
+    kicsDiagnosticCollection,
+    [],
+    []
+  );
+  const kicsScanCommand = new KICSRealtimeCommand(context, kicsProvider, logs);
+  kicsScanCommand.registerKicsScans();
+
+  const diagnosticCollection =
+    vscode.languages.createDiagnosticCollection(constants.extensionName);
+  const statusBarItem = vscode.window.createStatusBarItem(
+    vscode.StatusBarAlignment.Left
+  );
+  // Create listener for file saves for real time feedback
+  addRealTimeSaveListener(context, logs);
+  const filterCommand = new FilterCommand(context, logs);
+  const groupByCommand = new GroupByCommand(context, logs);
+  const astResultsProvider = new AstResultsProvider(
+    context,
+    logs,
+    statusBarItem,
+    diagnosticCollection,
+    filterCommand,
+    groupByCommand
+  );
+
+  // Syncing with AST everytime the extension gets opened
+  astResultsProvider
+    .openRefreshData()
+    .then(() => logs.info(messages.dataRefreshed));
+  // Initialize filters state
+  filterCommand
+    .initializeFilters()
+    .then(() => logs.info(messages.filtersInitialized));
+  // Initialize group by state
+  groupByCommand
+    .initializeFilters()
+    .then(() => logs.info(messages.groupByInitialized));
+  // Workspace listener
+  const workspaceListener: WorkspaceListener = new WorkspaceListener();
+  setInterval(
+    () => workspaceListener.listener(context, astResultsProvider),
+    500
+  );
+  // Results side tree creation
+  vscode.window.registerTreeDataProvider(constants.treeName, astResultsProvider);
+  const tree = vscode.window.createTreeView(constants.treeName, {
+    treeDataProvider: astResultsProvider,
+  });
+  // tree listener to open a webview in a new panel with results details
+  tree.onDidChangeSelection((item) => {
+    if (item.selection.length > 0) {
+      if (!item.selection[0].contextValue && !item.selection[0].children) {
+        // Open new details
+        vscode.commands.executeCommand(commands.newDetails, item.selection[0].result);
+      }
     }
-    ));
-
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.viewKicsSaveSettings`, () => {
+  });
+  // Webview detailsPanel to show result details on the side
+  const webViewCommand = new WebViewCommand(context, logs, astResultsProvider);
+  const newDetails = webViewCommand.registerNewDetails();
+  context.subscriptions.push(newDetails);
+  // Branch change Listener
+  await gitExtensionListener(context, logs);
+  // SCA Auto Scanning view
+  const scaResultsProvider = new SCAResultsProvider(
+    context,
+    logs,
+    statusBarItem,
+    diagnosticCollection
+  );
+  scaResultsProvider.scaResults = [];
```

Comment:
can we constructor this with an empty array. ?

Reply:
done